lib/cosmos/tools/table_manager/table_manager_core.rb

```diff
@@ -14,463 +14,269 @@
+        @config.tables.each do |table_name, table|
           file.write(table.buffer)
         end
       end
+      file_report(filename, @config.filename)
     end
 
+    # @return [String] Success string if parameters all check. Raises
+    #   a CoreError if errors are found.
     def file_check
-      raise "Please open a table first." unless @table_def
-
+      raise NoConfigError unless @config
       result = ""
-      @table_def.get_all_tables.each do |table|
-        table_result = table_check(table.name)
+      @config.table_names.each do |name|
+        table_result = table_check(name)
         unless table_result.empty?
-          result << "Error(s) in #{table.name}:\n" + table_result
+          result << "Errors in #{name}:\n" + table_result
         end
       end
-      result
-    end
-
-    def file_hex
-      raise "Please open a table first." unless @table_def
-
-      data = ""
-      # collect the data from each table
-      @table_def.get_all_tables.each do |table|
-        data << table.buffer
-      end
-
-      "#{data.formatted}\n\nTotal Bytes Read: #{data.length}"
+      raise CoreError, result unless result.empty?
+      "All parameters are within their constraints."
     end
 
-    # Generate the RPT file for the currently opened file
-    def file_report
-      raise "Please open a table first." unless @table_def
-
-      result = file_check()
-      unless result.empty?
-        raise "Please fix the following errors before generating the report:\n\n" << result
-      end
-
-      filename = File.basename(@current_bin, ".dat")
-      File.open(File.join(File.dirname(@current_bin), "#{filename}.rpt"), 'w+') do |file|
-        file.write("File Definition: #{@current_def}\n")
-        file.write("File Binary: #{@current_bin}\n\n")
-        @table_def.get_all_tables.each do |table|
+    # Create a CSV report file based on the file contents.
+    #
+    # @param [String] Binary filename currently open. Used to generate the
+    #   report name such that it matches the binary filename.
+    # @return [String] Report filename path
+    def file_report(bin_path, def_path)
+      raise NoConfigError unless @config
+      file_check()
+
+      basename = File.basename(bin_path, ".dat")
+      report_path = File.join(File.dirname(bin_path), "#{basename}.csv")
+      File.open(report_path, 'w+') do |file|
+        file.write("File Definition, #{def_path}\n")
+        file.write("File Binary, #{bin_path}\n\n")
+        @config.tables.values.each do |table|
           items = table.sorted_items
-          file.puts(table.name)
-
-          column_lengths = []
+          file.puts(table.table_name)
 
+          # Write the column headers
           if table.type == :TWO_DIMENSIONAL
-            column_lengths[0] = "Item".length
+            columns = ["Item"]
+            # Remove the '0' from the 'itemname0'
+            table.num_columns.times.each {|x| columns << items[x].name[0...-1] }
```

Comment:
This is kind of a hack for the two dimensional tables but it works and avoids other uglier hacks I tried.